### PR TITLE
Removes  "_ " in schema name for DM -schema family 

### PR DESCRIPTION
**Pull request by Kajal-NEC**
This pull request fixes issue #2201

### CODE REVIEW

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/utils/NGSIUtils.java [hunk 1]
```diff
@@ -60,7 +60,11 @@ private NGSIUtils() {
      */
     public static String encode(String in, boolean deleteSlash, boolean encodeSlash) {
         if (deleteSlash) {
-            return ENCODEPATTERN.matcher(in.substring(1)).replaceAll("_");
+           if(in.startsWith("/")) {
+                return ENCODEPATTERN.matcher(in.substring(1)).replaceAll("_");
+            }else {
+                return ENCODEPATTERN.matcher(in).replaceAll("_");
+            }
```
Comment (fgalan) [inline cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/utils/NGSIUtils.java:67]: Indent looks weird... Maybe better:

```suggestion
            if(in.startsWith("/")) {
                return ENCODEPATTERN.matcher(in.substring(1)).replaceAll("_");
            } else {
                return ENCODEPATTERN.matcher(in).replaceAll("_");
            }
```

Comment (Kajal-NEC) [reply cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/utils/NGSIUtils.java]: Resolved in b539526

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIPostgreSQLSink.java [hunk 2]
```diff
@@ -547,7 +547,7 @@ public String buildSchemaName(String service, String subService) throws CygnusBa
                 case DMBYENTITYDATABASESCHEMA:
```
Comment (fgalan) [inline cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIPostgreSQLSink.java:547]: An entry in CHANGES_NEXT_RELEASE file should be included. I'd suggest:

```
- [cygnus-ngsi] Removes "_ " in schema name for DM -schema family (#2201)
```

**Comment by fgalan** [comment]
> Hi @fgalan , I have resolved the suggestions. Please merge the PR If it seems ok.

A couple of new suggestion has been posted.

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIPostgreSQLSink.java (hunk 2 above)
Comment (Kajal-NEC) [reply cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIPostgreSQLSink.java:547]: Added in 59c4a86

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/utils/NGSIUtils.java [hunk 3]
```diff
@@ -60,7 +60,11 @@ private NGSIUtils() {
      */
     public static String encode(String in, boolean deleteSlash, boolean encodeSlash) {
         if (deleteSlash) {
-            return ENCODEPATTERN.matcher(in.substring(1)).replaceAll("_");
+            if(in.startsWith("/")) {
```
[the remark this answers is not in the document]
Comment (Kajal-NEC) [reply cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/utils/NGSIUtils.java]: Resoved in 2db6590

**Review by fgalan — APPROVED** [review]
LGTM

**Comment by fgalan** [comment]
Merging. Thanks for your contribution!

**Pull request merged**